pinot-tools/src/main/java/org/apache/pinot/tools/service/api/resources/PinotServiceManagerInstanceResource.java

```diff
@@ -0,0 +1,241 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.pinot.common.utils.CommonConstants;
+import org.apache.pinot.common.utils.NetUtil;
+import org.apache.pinot.controller.ControllerConf;
+import org.apache.pinot.spi.services.ServiceRole;
+import org.apache.pinot.spi.utils.JsonUtils;
+import org.apache.pinot.tools.service.PinotServiceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.pinot.common.utils.CommonConstants.Controller.CONFIG_OF_CONTROLLER_METRICS_PREFIX;
+import static org.apache.pinot.common.utils.CommonConstants.Controller.DEFAULT_METRICS_PREFIX;
+import static org.apache.pinot.tools.utils.PinotConfigUtils.TMP_DIR;
+import static org.apache.pinot.tools.utils.PinotConfigUtils.getAvailablePort;
+
+
+@Api(tags = "Startable")
+@Path("/")
+public class PinotServiceManagerInstanceResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotServiceManagerInstanceResource.class);
+
+  @Inject
+  private PinotServiceManager _pinotServiceManager;
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/instances")
+  @ApiOperation(value = "Get Pinot Instances Status")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Instance Status"), @ApiResponse(code = 500, message = "Internal server error")})
+  public Map<String, PinotInstanceStatus> getPinotAllInstancesStatus() {
+    Map<String, PinotInstanceStatus> results = new HashMap<>();
+    for (String instanceId : _pinotServiceManager.getRunningInstanceIds()) {
+      results.put(instanceId, _pinotServiceManager.getInstanceStatus(instanceId));
+    }
+    return results;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/instances/{instanceName}")
+  @ApiOperation(value = "Get Pinot Instance Status")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Instance Status"), @ApiResponse(code = 404, message = "Instance Not Found"), @ApiResponse(code = 500, message = "Internal server error")})
+  public PinotInstanceStatus getPinotInstanceStatus(
+      @ApiParam(value = "Name of the instance") @PathParam("instanceName") String instanceName) {
+    List<String> instanceIds = _pinotServiceManager.getRunningInstanceIds();
+    if (instanceIds.contains(instanceName)) {
+      return _pinotServiceManager.getInstanceStatus(instanceName);
+    }
+    throw new WebApplicationException(String.format("Instance [%s] not found.", instanceName),
+        Response.Status.NOT_FOUND);
+  }
+
+  @DELETE
```

Comment:
is it possible to add "stopAllServers" "stopAllBrokers" etc, and "stopAll"

Reply:
added `stopAllPinotInstances` and `stopPinotInstancesByRole`